mkdcomments.py

```diff
@@ -58,7 +55,25 @@ def _unmultiliner(self, line):
 
 class RawCommentReplacer(Postprocessor):
     def run(self, text):
-        return re.sub(PREFIX_PLACEHOLDER, '<!---', text)
+        # Restore the placeholder
+        text = re.sub(PREFIX_PLACEHOLDER, '<!---', text)
+
+        # Remove unwanted <p> tags around list items.
+        # See issue #8 for details.
+        text = re.sub(r'<li>\s*<p>(.*?)</p>\s*</li>', r'<li>\1</li>', text)
```

Comment:
Per discussion in #8, the thing that makes me uncomfortable with this solution is that I believe this will "correct" undesirable output caused by blank lines and not just by comments. IMO that would not be a great side-effect because it would encourage people to write bad markdown by hiding their mistakes from them.